src/VisualStudio/TestUtilities/IntegrationHost.cs

```diff
@@ -0,0 +1,323 @@
+
+        ~IntegrationHost()
+        {
+            Dispose(false);
+        }
+
+        public DTE Dte
+            => _dte;
+
+        public InteractiveWindow CSharpInteractiveWindow
+        {
+            get
+            {
+                if (_csharpInteractiveWindow == null)
+                {
+                    _csharpInteractiveWindow = InteractiveWindow.CreateCSharpInteractiveWindow(this);
+                }
+
+                return _csharpInteractiveWindow;
+            }
+        }
+
+        public EditorWindow EditorWindow
+        {
+            get
+            {
+                if (_editorWindow == null)
+                {
+                    _editorWindow = new EditorWindow(this);
+                }
+
+                return _editorWindow;
+            }
+        }
+
+        public bool RequireNewInstance
+            => (_hostProcess == null) || _hostProcess.HasExited || _requireNewInstance;
+
+        public SolutionExplorer SolutionExplorer
+        {
+            get
+            {
+                if (_solutionExplorer == null)
+                {
+                    _solutionExplorer = new SolutionExplorer(this);
+                }
+
+                return _solutionExplorer;
+            }
+        }
+
+        public void Cleanup()
+        {
+            CleanupDte();
+
+            if (RequireNewInstance)
+            {
+                IntegrationLog.Current.WriteLine("Closing existing Visual Studio instance.");
+
+                CleanupRemotingService();
+                CleanupHostProcess();
+            }
+        }
+
+        public void ClickAutomationElement(string elementName, bool recursive = false)
+            => ClickAutomationElementAsync(elementName, recursive).ConfigureAwait(continueOnCapturedContext: false).GetAwaiter().GetResult();
+
+        public async Task ClickAutomationElementAsync(string elementName, bool recursive = false)
+        {
+            var automationElement = await LocateAutomationElementAsync(elementName, recursive).ConfigureAwait(continueOnCapturedContext: false);
+
+            object invokePattern = null;
+            if (automationElement.TryGetCurrentPattern(InvokePattern.Pattern, out invokePattern))
+            {
+                ((InvokePattern)(invokePattern)).Invoke();
+            }
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            // Explicitly don't call GC.SuppressFinalize() so that the finalizer is still called and the _hostProcess is killed
+        }
+
+        public void Initialize()
+        {
+            Cleanup();
+
+            if (RequireNewInstance)
+            {
+                IntegrationLog.Current.WriteLine("Starting a new instance of Visual Studio.");
+
+                InitializeHostProcess();
+                InitializeDte();
+                InitializeRemotingService();
+            }
+        }
+
+        internal async Task ExecuteDteCommandAsync(string command, string args = "")
```

Comment:
I wasn't use this as 'good' API practice. I was doing it because this just passes the args into `Dte.ExecuteCommand` and if I make my default 'null', then it automatically changes the behavior of what calling `Dte.ExecuteCommand` directly would do.